single_stage_detector/README.md

```diff
@@ -0,0 +1,293 @@
+convert the labels to [COCO](https://cocodataset.org/#home) format. The
+downloaded dataset size is 352GB and the expected folder structure after
+running the script is:
+```
+<DOWNLOAD_PATH>
+│
+└───info.json
+│
+└───train
+│   └─── data
+│   │      000002b66c9c498e.jpg
+│   │      000002b97e5471a0.jpg
+│   │      ...
+│   └─── metadata
+│   │      classes.csv
+│   │      hierarchy.json
+│   │      image_ids.csv
+│   └─── labels
+│          detections.csv
+│          openimages-mlperf.json
+│
+└───validation
+    └─── data
+    │      0001eeaf4aed83f9.jpg
+    │      0004886b7d043cfd.jpg
+    │      ...
+    └─── metadata
+    │      classes.csv
+    │      hierarchy.json
+    │      image_ids.csv
+    └─── labels
+           detections.csv
+           openimages-mlperf.json
+```
+
+Read more about the mlperf subset [here](#the-mlperf-subset).
+
+
+## Download the pretrained backbone
+The benchmark uses a ResNeXt50_32x4d backbone pretrained on ImageNet. The
+weights are downloaded from PyTorch hub.
+
+By default, the code will automatically download the weights to
+`$TORCH_HOME/hub` (default is `~/.cache/torch/hub`) and save them for later use.
+
+Alternatively, you can manually download the weights with:
+```bash
+bash ./single_stage_detector/scripts/download_backbone.sh
+```
+
+Then use the downloaded file with `--pretrained <PATH TO WEIGHTS>` .
+
+If you wish to use the backbone in frameworks other than PyTorch, the repository
+provides scripts to convert the weights to ONNX format and a pickled dictionary
+of numpy arrays:
+```
+./single_stage_detector/scripts/pth_to_onnx.py --help
+./single_stage_detector/scripts/pth_to_pickle.py --help
+```
+
+## Training on NVIDIA DGX-A100 (single node)
+Launch configuration and system-specific hyperparameters for the NVIDIA DGX-A100
+single node reference are in the `config_DGXA100_001x08x032.sh` script.
+
+Steps required to launch single node training on NVIDIA DGX-A100:
+
+```
+cd ./single_stage_detector/ssd
+source config_DGXA100_001x08x032.sh
+DATADIR=<path/to/data/dir> LOGDIR=<path/to/output/dir> ./run.sub
+```
+
+## Training on NVIDIA DGX-A100 (multi node)
+Launch configuration and system-specific hyperparameters for the NVIDIA DGX-A100
+multi node reference are in the `config_DGXA100_008x08x008.sh` and
+`config_DGXA100_032x08x032.sh` scripts.
+
+Steps required to launch multi node training on NVIDIA DGX-A100:
+
+```
+cd ./single_stage_detector/ssd
+source config_DGXA100_008x08x008.sh # or config_DGXA100_032x08x032.sh
+DATADIR=<path/to/data/dir> LOGDIR=<path/to/output/dir> sbatch -N $DGXNNODES -t $WALLTIME run.sub
+```
+
+## Hyperparameter settings
+Hyperparameters are recorded in the `config_*.sh` files for each configuration
+and in `run_and_time.sh`.
+
+# Dataset/Environment
+## Publication/Attribution
+[Google Open Images Dataset V6](https://storage.googleapis.com/openimages/web/index.html)
+
+## The MLPerf Subset
+The MLPerf subset includes only 264 classes of the total 601 available in the
```

Comment:
I think it would be useful to include here how these 264 classes were chosen....

Reply:
done